[UIK-2455][filter trigger] a11y (#1786)
semrush/intergalactic: semcore/base-trigger/CHANGELOG.md

CHANGELOG.md standards are inspired by [keepachangelog.com](https://keepachangelog.com/en/1.0.0/).

## [4.49.0] - 2024-11-20

### Fixed

* Added `Hint` to the `FilterTrigger.ClearButton`.
* Remove redundant `aria-label` & `aria-labelledby` from `FilterTrigger.ClearButton`.
* Added `role` combobox to the `FilterTrigger.TriggerButton`.

> * Added `count` prop to `FilterTrigger.Counter`, along with screenreader only `selected` text

Reviewer: let's also mention that placeholder text is now aria-hidden

Author: added

CHANGELOG.md standards are inspired by [keepachangelog.com](https://keepachangelog.com/en/1.0.0/).

## [4.49.0] - 2024-11-20

### Fixed

- Added `Hint` to the `FilterTrigger.ClearButton`.
- Remove redundant `aria-label` & `aria-labelledby` from `FilterTrigger.ClearButton`.
- Added `role` combobox to the `FilterTrigger.TriggerButton`.

> - Added screenreader only `selected` text to the `FilterTrigger.Counter` value.

Reviewer: ```suggestion
- Added `count` prop to `FilterTrigger.Counter`, along with screenreader only `selected` text
```
to mention the new prop as well

CHANGELOG.md standards are inspired by [keepachangelog.com](https://keepachangelog.com/en/1.0.0/).

## [4.49.0] - 2024-11-20

### Fixed

- Added `Hint` to the `FilterTrigger.ClearButton`.
- Remove redundant `aria-label` & `aria-labelledby` from `FilterTrigger.ClearButton`.
- Added `role` combobox to the `FilterTrigger.TriggerButton`.
- Added `count` prop to `FilterTrigger.Counter`, along with screenreader only `selected` text.

> - `placeholder` text is now `aria-hidden` for the `FilterTrigger.TriggerButton` because of `aria-label` and double pronouncing.

Reviewer: sorry for clinging to this, but maybe better not mention specifically FilterTrigger.TriggerButton? People might think it's only relevant for that component, when it's also relevant for Select.Trigger and other BaseTrigger-based components, if I understand correctly

Author: fixed
